### PR TITLE
docs/examples: document traffic policy modes and update example workflow

### LINKED ISSUE #1528: Manual demo documentation for permissive mode
**Please describe what should be documented
The manual demo doc does not have the examples/manifest files to demo permissive mode. With permissive mode, we need one of the following:
1. Tweak bookbuyer and bookthief apps to make requests to a service backed by pods. By default client apps make requests to the `bookstore` service, but we don't have pods backing this service.
2. Create a manifest for a service `bookstore` with Deployments.

**Please suggest where in the repo the document should be located
https://github.com/openservicemesh/osm/blob/main/docs/example/README.md

### PR DESCRIPTION
**Pull request by shashankram**
**Description**:
This change updates the example readme to document the usage of
traffic policy modes (SMI vs permissive) and reworks the example
demo for simplified onboarding. The existing demo is set up in
a way that works but is hard to comprehend for new users.

- Adds doc section for traffic policy modes
- Demo with permissive mode enabled
- Demo with SMI traffic policies
- manifest consolidation into appropriate folders

Resolves #1528

Signed-off-by: Shashank Ram <shashr2204@gmail.com>

**Affected area**:

- New Functionality      [ ]
- Documentation          [X]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`No`

### CODE REVIEW

File: docs/example/README.md [hunk 1]
```diff
@@ -31,25 +35,40 @@ The OSM Manual Install Demo Guide is designed to quickly allow you to demo and e
...
-The `Bookstore`, `Bookbuyer`, `Bookthief`, `Bookwarehouse` demo applications will be installed in their respective Kubernetes Namespaces. In order for these applications to be injected with a Envoy sidecar automatically, we must add the Namespaces to be monitored by the mesh.
+The demo consists of the following resources:
+- `bookbuyer` application that makes requests to the `bookstore` service to buy books
+- `bookthief` application that makes requests to the `bookstore` service to steal books
+- `bookstore` service that allows clients to purchase books
+- `bookwarehouse` service that the `bookstore` service reaches out to restock books
+
+When we demonstrate traffic splitting using SMI Traffic Split, we will deploy two additional services:
+- `bookstore-v1` service representing version v1 of the `bookstore` service
+- `bookstore-v2` service representing version v1 of the `bookstore` service
```
Comment (ritazh) [inline docs/example/README.md:69]:
```suggestion
- `bookstore-v2` service representing version v2 of the `bookstore` service
```

File: docs/example/README.md [hunk 2]
```diff
@@ -176,44 +283,43 @@ spec:
...
 ```bash
-kubectl apply -f docs/example/manifests/split-v2/
+kubectl apply -f docs/example/manifests/split/traffic-split-v2.yaml
 ```
 
 Wait for the changes to propagate and observe the counters increment for bookstore-v2 in your browser windows. Modify the `weight` fields in [manifests/split-v2/traffic-split-v2.yaml](manifests/split-v2/traffic-split-v2.yaml) and re-apply changes to experiment.
 
 - http://localhost:8082 - **bookstore-v2**
 
+## Traffic Encryption
```
Comment (ritazh) [inline docs/example/README.md:319]: should we move this up right after `Traffic Policy Modes`?

File: docs/example/README.md [hunk 3]
```diff
@@ -31,25 +35,40 @@ The OSM Manual Install Demo Guide is designed to quickly allow you to demo and e
   - ```git clone https://github.com/openservicemesh/osm.git```
   - ```cd osm```
 
-
-
 ## Install OSM CLI
 Use the [installation guide](/docs/installation_guide.md) to install the `osm` cli.
```
Comment (ritazh) [inline docs/example/README.md:39]: broken link

File: docs/example/README.md [hunk 4]
```diff
@@ -31,25 +35,40 @@ The OSM Manual Install Demo Guide is designed to quickly allow you to demo and e
...
+1.  Install OSM in SMI traffic policy mode:
     ```bash
     osm install
     ```
+
 1. To enable Prometheus and Grafana, use their respective flags
     ```bash
     osm install --deploy-prometheus true --deploy-grafana true
     ```
     See the [metrics documentation](/docs/patterns/observability/metrics.md#automatic-bring-up) for more details.
```
Comment (ritazh) [inline docs/example/README.md:58]: broken link

**Comment by ritazh** [comment]
Went thru this PR with v0.7.0-rc.1. The demo flow works as expected. Thanks for the PR!

**Comment by mayurigupta13** [comment]
Under the header of 'Install OSM control plane' and bullet 1, we should specify that Prometheus and Grafana are not installed when no flags are specified. This was mentioned earlier.

**Comment by shashankram** [comment]
> Under the header of 'Install OSM control plane' and bullet 1, we should specify that Prometheus and Grafana are not installed when no flags are specified. This was mentioned earlier.

Done, I added a note.

**Comment by shashankram** [comment]
> Went thru this PR with v0.7.0-rc.1. The demo flow works as expected. Thanks for the PR!

Thanks for providing feedback and testing the updated workflow @ritazh

File: docs/example/manifests/access/traffic-access-v1.yaml [hunk 5]
```diff
@@ -19,8 +19,8 @@ spec:
     name: bookbuyer
     namespace: bookbuyer
   #- kind: ServiceAccount
-    #name: bookthief
-    #namespace: bookthief
+     #name: bookthief
```
Comment (ritazh) [inline docs/example/manifests/access/traffic-access-v1.yaml:22]: I tested this change with command+/, which resolves to an extra space. Reverting this update works as expected.

**Review by ritazh — APPROVED** [review]

**Review by eduser25 — APPROVED** [review]

**Review by draychev — APPROVED** [review]

**Review by michelleN — APPROVED** [review]

**Pull request merged**